src/Models/Cart.php

```diff
@@ -98,7 +115,11 @@ public function processOrder(Model $payment): Order
         $payment->order_id = $order->id;
         $payment->save();
 
-        $this->markVouchersAsUsed();
+        static::activeDeductions()
+            ->each(function (CartDeduction $deduction) {
+                $deduction::markCartDeductionsAsUsed($this);
+            });
```

Comment:
Example of handling all CartDeductions the same way.  "active" means they have bindings registered in the container - you'll see this later in the PR.